### DISCUSSION TITLE
Custom snippet and pagelist

### DISCUSSION THREAD

**Discussion by cyril-montana** [automadcms/discussions#7] [General]
Hi Marc
I'm using a custom snippet in one of my pages.
This snippet looks good while I'm looking at the page preview.
But totaly disabled when the page is part of a pagelist. 
It looks like the pagelist cannot support a custom snippet.

Sorry for my dirty english.
Thanks for helping :)

**Comment by mdahmenzpf** [comment]
What version do you use? Is caching disabled? What does the snippet look like?

**Comment by cyril-montana** [comment]
Hello Marc
I'm using V.1 and caching is enabled. I'm using the same snippet (same variables but differents values) in some pages.
First, I'm using simple html in the snippet (<div class="title">value 01</div>). It works well in the single page but not appears when insert the page in a pagelist (based on skel.

Then, using shared variables in the snippet :
```
<@ set { %variable: 'value 01' } @> 
<div class="title">@{ %variable }</div>
```
It looks well in the single page, but not in the pagelist…

Thanks for helping :)

**Comment by marcantondahmen** [comment]
What do you want to achieve with that snippet? Why do want to set session variables in a loop?

In general, pagelist snippets work very differently. You should not use the same snippet in a pagelist and in a snippet block. A pagelist snippet only renders the current page in a loop. It should not contain other logic.

Take a look at the snippets that are included with the standard theme.

**Reply by cyril-montana** [reply]
Ok, I know understand :)
I wanted to use pages as cards. Reinterpret each items in the page (header, title, content, footer) with specific classes.
(probably something approaching the trick you use for your own website)
```
<div class="header">text</div>
<div class="title">title</div>
<div class="decription">desc</div>
```
I've find an alternative interesting with the markdown tool.

**Reply by marcantondahmen** [reply]
Yeah, I guess you're on the right track. You can create cards like that. Just make sure you only include the actual pagelist loop without the config. In other words, just the `foreach` statement including the card markup.
